Week3/index.js

```diff
@@ -0,0 +1,249 @@
+        todo_desc,
+        listIdValue,
+        userIdValue,
+      ]);
+      await execQuery(sql);
+      response.send(201, 'item added successfully');
+    }
+  } catch (error) {
+    console.error(error);
+    response.status(400)('something went wrong');
+  }
+});
+
+app.delete('/todo/:user_name/:list_name', async (request, response) => {
+  try {
+    const user_name = request.params.user_name;
+    const list_name = request.params.list_name;
+    const { todo_desc } = request.body;
+    const userIdValue = await idToValue('user_id', 'user', 'user_name', user_name);
+    const listIdValue = await idToValue('list_id', 'todo_list', 'list_name', list_name);
+    if (listIdValue == '' || userIdValue == '') {
+      response.status(400).send('user name or list name does not exist');
+    } else {
+      const sql = createQuery([
+        'DELETE FROM todo_item WHERE user_id = ? AND list_id = ? AND todo_desc = ?',
+        userIdValue,
+        listIdValue,
+        todo_desc,
+      ]);
+
+      await execQuery(sql);
+      response.send(200, 'item deleted successfully');
+    }
+  } catch (error) {
+    console.error(error);
+    response.status(400).send('something went wrong');
+  }
+});
+
+app.delete('/todo_list/:user_name/:list_name', async (request, response) => {
+  try {
+    const user_name = request.params.user_name;
+    const list_name = request.params.list_name;
+    const userIdValue = await idToValue('user_id', 'user', 'user_name', user_name);
+    const listIdValue = await idToValue('list_id', 'todo_list', 'list_name', list_name);
+    if (listIdValue == '' || userIdValue == '') {
+      response.status(400).send('user name or list name does not exist');
+    } else {
+      const sqlDeleteItems = createQuery([
+        'DELETE FROM todo_item WHERE user_id = ? AND list_id = ?',
+        userIdValue,
+        listIdValue,
+      ]);
+      const sql = createQuery([
+        'DELETE FROM todo_list WHERE user_id = ? AND list_id = ?',
+        userIdValue,
+        listIdValue,
+      ]);
+      await execQuery(sqlDeleteItems);
+      await execQuery(sql);
+      response.send(200, 'item deleted successfully');
+    }
+  } catch (error) {
+    console.error(error);
+    response.status(400).send('something went wrong');
+  }
+});
+
+app.put('/todo/done/:user_name/:list_name', async (request, response) => {
+  try {
+    const user_name = request.params.user_name;
+    const list_name = request.params.list_name;
+    if (listIdValue == '' || userIdValue == '') {
+      response.status(400).send('user name or list name does not exist');
+    } else {
+      const { todo_desc } = request.body;
+      const userIdValue = await idToValue('user_id', 'user', 'user_name', user_name);
+      const listIdValue = await idToValue('list_id', 'todo_list', 'list_name', list_name);
+      const sql = createQuery([
+        'UPDATE `todo_item` SET `is_done` = 1 WHERE `user_id` = ? AND `list_id` = ? AND todo_desc = ?',
```

Comment:
I am curious why here you are not checking if userId and listId are valid? It seems to me that it would be easy to forget that the function idToValue returns an empty string when the id is not found? This could lead to some nasty bugs. Best not to leave such things to chance. How could you ensure that no one makes a mistake by being careless like that?